modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,366 @@ redirect_from:
 
+These effects tend to show up in functions working on a single piece of data - for instance parsing a single `String` into an `Int`, validating a login, or asynchronously fetching website information for a user.
+
+```kotlin:ank
+import arrow.core.Either
+import arrow.core.Option
+import arrow.fx.Promise
+
+interface Profile
+interface User
+
+fun parseInt(s: String): Option<Int> = TODO()
+data class SecurityError(val message: String)
+data class Credentials(val email: String, val password: String)
+
+fun validateLogin(c: Credentials): Either<SecurityError, Unit> = TODO()
+
+fun userInfo(u: User): Promise<ForIO, Profile> = TODO()
+```
+
+Each function asks only for the data it actually needs; in the case of `userInfo`, a single `User`. We certainly could write one that takes a `List<User>` and fetch profile for all of them, would be a bit strange. If we just wanted to fetch the profile of just one user, we would either have to wrap it in a `List` or write a separate function that takes in a single user anyways. More fundamentally, functional programming is about building lots of small, independent pieces and composing them to make larger and larger pieces - does this hold true in this case?
+
+Given just `(User) -> Promise<ForIO, Profile>`, what should we do if we want to fetch profiles for a `List<User>`? We could try familiar combinators like map.
+
+```kotlin:ank
+fun profilesFor(users: List<User>): List<Promise<ForIO, Profile>> = users.map(::userInfo)
+```
+
+Note the return type `List<Promise<ForIO, Profile>>`. This makes sense given the type signatures, but seems unwieldy. We now have a list of asynchronous values, and to work with those values we must then use the combinators on `Promise` for every single one. It would be nicer instead if we could get the aggregate result in a single `Promise`, say a `Promise<ForIO, List<Profile>>`.
+
+There exists a much more generalized form that would allow us to parse a `List<String>` or validate credentials for a `List<User>`.
+
+Enter `Traverse`.
+
+### The Typeclass
+At center stage of Traverse is the traverse method.
+
+```kotlin
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Functor
+
+interface Traverse<F> : Functor<F>, Foldable<F> {
+  fun <G, A, B> Kind<F, A>.traverse(AP: Applicative<G>, f: (A) -> Kind<G, B>): 
+}
+```
+
+**Essentially, one uses `Traverse`, when there is a function `(A) -> Kind<G, B>` and `G` is an `Applicative` instance e.g.: `(User) -> Promise<ForIO, Profile>` and you want to `lift` the previous function to work on `Kind<F, A>` where `F` is a `Traverse` instance e.g.: `List<User>` and return `Kind<G, Kind<F, B>>` e.g.: `Promise<ForIO, List<Profile>>`**
```

Comment:
This paragraph is too dense. I would split it.

Reply:
I deleted this section because the next paragraph explains the same concept, just with more concrete Datatypes. I am shifting away from this style of writing.